share/tools/web_config/sample_prompts/sorin.fish

```diff
@@ -1,61 +1,219 @@
-                    printf (set_color blue)✱' '
-                case "*R*"
-                    printf (set_color brmagenta)➜' '
-                case "*U*"
-                    printf (set_color bryellow)═' '
-                case "??"
-                    printf (set_color brwhite)◼' '
+    # Count added, deleted, modified, renamed, unmerged, and untracked.
+    # T (type change) is undocumented, see http://git.io/FnpMGw.
+    # For a table of scenarii, see http://i.imgur.com/2YLu1.png.
+    set -l status_added 0
+    set -l status_deleted 0
+    set -l status_modified 0
+    set -l status_renamed 0
+    set -l status_unmerged 0
+    set -l status_untracked 0
+    for line in (command git status --porcelain | cut -b-2)
+        # Check unambiguous cases first which allows us
+        # to skip running all the other regexps.
+        if test "$line" = '??'
+            set status_untracked 1
+            continue
+        end
+        if string match -r '^(?:AA|DD|U.|.U)$' "$line" >/dev/null
+            set status_unmerged 1
+            continue
+        end
+        if string match -r '^(?:[ACDMT][ MT]|[ACMT]D)$' "$line" >/dev/null
+            set status_added 1
+        end
+        if string match -r '^[ ACMRT]D$' "$line" >/dev/null
+            set status_deleted 1
+        end
+        if string match -r '^.[MT]$' "$line" >/dev/null
+            set status_modified 1
+        end
+        if string match -e 'R' "$line" >/dev/null
+            set status_renamed 1
+        end
+    end
+
+    set_color -o
+
+    if test -n "$branch"
+        if test $branch_detached -ne 0
+            set_color brmagenta
+        else
+            set_color green
+        end
+        echo -n " $branch"
+    end
+    if test -n "$commit"
+        echo -n ' '(set_color yellow)"$commit"
+    end
+    if test -n "$action"
+        set_color normal
+        echo -n (set_color white)':'(set_color -o brred)"$action"
+    end
+    if test $status_ahead -ne 0
+        echo -n ' '(set_color brmagenta)'⬆'
+    end
+    if test $status_behind -ne 0
+        echo -n ' '(set_color brmagenta)'⬇'
+    end
+    if test $status_stashed -ne 0
+        echo -n ' '(set_color cyan)'✭'
+    end
+    if test $status_added -ne 0
+        echo -n ' '(set_color green)'✚'
+    end
+    if test $status_deleted -ne 0
+        echo -n ' '(set_color red)'✖'
+    end
+    if test $status_modified -ne 0
+        echo -n ' '(set_color blue)'✱'
+    end
+    if test $status_renamed -ne 0
+        echo -n ' '(set_color magenta)'➜'
+    end
+    if test $status_unmerged -ne 0
+        echo -n ' '(set_color yellow)'═'
+    end
+    if test $status_untracked -ne 0
+        echo -n ' '(set_color white)'◼'
+    end
+
+    set_color normal
+end
+
+function _git_action --argument-names git_dir
+    for action_dir in "$git_dir/rebase-apply" "$git_dir/rebase"
+        if test -d "$action_dir"
+            if test -f "$action_dir/rebasing"
+                echo -n 'rebase'
```

Comment:
The quotes here around literals are unnecessary. Not that they hurt, of course.

Reply:
Would you like me to remove them?